workers/social/lib/social/models/account.coffee

```diff
@@ -367,6 +373,38 @@ module.exports = class JAccount extends jraphical.Module
           callback null, groups
 
 
+  fetchInviteGroups$: secure (client, options, callback) ->
+
+    @fetchInviteGroups options, callback
+
+
+  fetchInviteGroups: (options, callback) ->
+
+    [ options, callback ] = [ callback, options ]  unless callback
+    options ?= {}
+
+    queue = [
+      (next) =>
+        @fetchEmail next
+      (email, next) ->
+        JInvitation = require './invitation'
+        JInvitation.some { email, status : 'pending' }, options, next
+      (invitations, next) ->
+        { uniq, map } = _
+        slugs  = uniq map invitations, (invitation) -> invitation.groupName
+        JGroup = require './group'
+        JGroup.some { slug : { $in : slugs } }, {}, (err, groups) ->
```

Comment:
better to provide a `limit: 10` here into `options` or completely changing this into an iterator like here https://github.com/koding/koding/blob/master/workers/social/lib/social/models/emailnotification.coffee#L84-L100 (just as an example usage of `someData` and `cursor`)